test_scripts/Resumption/Handling_errors_from_HMI/009_restore_subscription_for_app2_after_fail_for_app1.lua

```diff
@@ -0,0 +1,86 @@
+-- 3. respond RegisterAppInterfaceResponse(success=true,result_code=RESUME_FAILED) to mobile application app1
+-- 4. restore all data for app2 and respond RegisterAppInterfaceResponse(success=true,result_code=SUCCESS)to mobile application app2
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/Resumption/Handling_errors_from_HMI/commonResumptionErrorHandling')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+-- [[ Local Variables ]]
+local vehicleDataSpeed = {
+  requestParams = { speed = true },
+  responseParams = { speed = { resultCode = "SUCCESS", dataType = "VEHICLEDATA_SPEED"} }
+}
+
+local vehicleDataRpm = {
+  requestParams = { rpm = true },
+  responseParams = { rpm = { resultCode = "SUCCESS", dataType = "VEHICLEDATA_RPM"} }
+}
+
+-- [[ Local Function ]]
+local function checkResumptionData()
+  common.getHMIConnection():ExpectRequest("VehicleInfo.SubscribeVehicleData")
+  :Do(function(exp, data)
+      if exp.occurences == 1 or
+      exp.occurences == 2 and
+      data.params.gps then
+        local function sendResponse()
+          common.getHMIConnection():SendError(data.id, data.method, "GENERIC_ERROR", "info message")
+        end
+        RUN_AFTER(sendResponse, 300)
+      else
+        common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+      end
+    end)
+  :Times(4)
+
+  common.getHMIConnection():ExpectRequest("VehicleInfo.UnsubscribeVehicleData", vehicleDataSpeed.requestParams)
+  :Do(function(_,data)
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+
+runner.Title("Test")
+runner.Step("Register app1", common.registerAppWOPTU)
+runner.Step("Register app2", common.registerAppWOPTU, { 2 })
+runner.Step("Activate app1", common.activateApp)
+runner.Step("Activate app2", common.activateApp, { 2 })
+runner.Step("Add for app1 subscribeVehicleData gps", common.subscribeVehicleData)
+runner.Step("Add for app1 subscribeVehicleData speed", common.subscribeVehicleData, { 1, vehicleDataSpeed })
+runner.Step("Add for app2 subscribeVehicleData gps", common.subscribeVehicleData, { 2, nil, 0 })
+runner.Step("Add for app2 subscribeVehicleData fuelLevel", common.subscribeVehicleData, { 2, vehicleDataRpm })
+runner.Step("Unexpected disconnect", common.unexpectedDisconnect)
+runner.Step("Connect mobile", common.connectMobile)
+runner.Step("openRPCserviceForApp1", common.openRPCservice, { 1 })
+runner.Step("openRPCserviceForApp2", common.openRPCservice, { 2 })
+runner.Step("Reregister Apps resumption ", common.reRegisterApps, { checkResumptionData })
+
```

Comment:
@GetmanetsIrina Please add additional step with `onVehicleData` notification as in `010` script